compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -466,12 +470,29 @@ fn check_opaque_meets_bounds<'tcx>(
     match origin {
         // Checked when type checking the function containing them.
         hir::OpaqueTyOrigin::FnReturn(..) | hir::OpaqueTyOrigin::AsyncFn(..) => {}
+        // Nested opaque types occur only in associated types:
+        // ` type Opaque<T> = impl Trait<&'static T, AssocTy = impl Nested>; `
+        // They can only be referenced as `<Opaque<T> as Trait<&'static T>>::AssocTy`.
+        // We don't have to check them here because their well-formedness follows from the WF of
+        // the projection input types in the defining- and use-sites.
+        hir::OpaqueTyOrigin::TyAlias
+            if let DefKind::OpaqueTy = tcx.def_kind(tcx.parent(def_id.to_def_id())) => {}
```

Comment:
that makes me a bit unhappy. Can you not instead add the type of the parent opaque type as an assumed wf type?

Reply:
We should not only include the type of the parent opaque type in assumed_wf_types, but also all other types that appear in the trait inputs (e.g `&'static T` in `impl Trait<&'static T, Assoc=impl Nested>`). The implementation can get easily complicated if we consider multiple levels of nesting and binders like:
```rust
impl Fn<(&'static T,), Output = impl for<'a> Fn<(&'a T,), Output = impl Sized + 'a>>
```

I think the current implementation strikes a balance between simplicity and completeness. That being said, if you feel strongly about it, I'm happy to implement the limited version of including only the parent opaque type. Note that we would reject the test `wf-nested.rs`.